ansible_ai_connect/users/management/commands/generate_users_trials_reports.py

```diff
@@ -0,0 +1,111 @@
+#  Copyright Red Hat
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from datetime import datetime
+from typing import Optional, cast
+
+from dateutil.relativedelta import relativedelta
+from django.apps import apps
+from django.core.management.base import BaseCommand, CommandError
+from django.utils import timezone
+
+from ansible_ai_connect.users.reports.generators import (
+    UserMarketingReportGenerator,
+    UserTrialsReportGenerator,
+)
+from ansible_ai_connect.users.reports.postman import Report, Reports
+
+
+class Command(BaseCommand):
+
+    @staticmethod
+    def generate_user_trials_report(
+        plan_id: Optional[int] = None,
+        created_after: Optional[datetime] = None,
+        created_before: Optional[datetime] = None,
+    ) -> str:
+        generator = UserTrialsReportGenerator()
+        return generator.generate(plan_id, created_after, created_before)
+
+    @staticmethod
+    def generate_user_marketing_report(
+        plan_id: Optional[int] = None,
+        created_after: Optional[datetime] = None,
+        created_before: Optional[datetime] = None,
+    ) -> str:
+        generator = UserMarketingReportGenerator()
+        return generator.generate(plan_id, created_after, created_before)
+
+    def add_arguments(self, parser):
+        parser.add_argument("--dry-run", action="store_true", help="Do nothing", default=False)
+        parser.add_argument("--plan-id", help="Trail plan_id", type=int)
+        parser.add_argument(
+            "--created-after",
+            help="UTC formatted datetime after which trials were accepted.",
+            type=datetime,
+        )
+        parser.add_argument(
+            "--created-before",
+            help="UTC formatted datetime before which trials were accepted.",
+            type=datetime,
+        )
+        parser.add_argument(
+            "--auto-range",
+            action="store_true",
+            help="Uses date range based on execution time",
+            default=False,
+        )
+
+    def handle(self, dry_run, plan_id, created_after, created_before, auto_range, *args, **options):
+        plan_id = int(plan_id) if plan_id else None
+        created_after = created_after if created_after else None
+        created_before = created_before if created_before else None
+
+        if auto_range:
+            created_before = timezone.now()
+            created_after = created_before - relativedelta(days=1)
```

Comment:
IIRC, cron job doesn't guarantee to run exactly on the specified time due to scheduling/load etc.

So using this, there is possibly we will miss a small window in-between 2 daily runs.

Reply:
`--auto-range` now defaults the time elements to "midnight".

For example, `created_before=2024-12-31 00:00:00`, `created_after=2024-12-30 00:00:00`.